### PR TITLE
Fix exception serialization  in user deletion

### PR DESCRIPTION
**Pull request by marcospri**
Fixes: https://hypothesis.sentry.io/issues/4905837844/?project=37293&referrer=github-pr-bot

Took the opportunity to refactor some tests around this file.

### CODE REVIEW

File: h/views/admin/users.py [hunk 1]
```diff
@@ -144,8 +144,8 @@ def users_delete(request):
 
 
 @view_config(context=UserNotFoundError)
-def user_not_found(exc, request):  # pragma: no cover
-    request.session.flash(Markup(_(exc.message)), "error")
```
Comment (marcospri) [inline h/views/admin/users.py:148]: This is the only non-test code change in the PR.

File: tests/unit/h/views/admin/users_test.py [hunk 2]
```diff
@@ -75,19 +76,6 @@ def test_users_index_strips_spaces(models, pyramid_request):
     models.User.get_by_username.assert_called_with(pyramid_request.db, "bob", "foo.org")
 
 
-@users_index_fixtures
```
Comment (marcospri) [inline tests/unit/h/views/admin/users_test.py:78]: Merged some tests in this file, they were one-assertion-per-test style which becomes verbose pretty quickly. Done in a separate commit.

File: tests/unit/h/views/admin/users_test.py [hunk 3]
```diff
@@ -204,6 +153,13 @@ def test_users_delete_user_not_found_error(user_service, pyramid_request):
         users_delete(pyramid_request)
 
 
+def test_user_not_found_view(pyramid_request):
```
Comment (marcospri) [inline tests/unit/h/views/admin/users_test.py:156]: New test for previously un-tested view.

**Review by seanh — APPROVED** [review]
